enricher/api/src/main/java/io/fabric8/maven/enricher/api/BaseEnricher.java

```diff
@@ -133,8 +134,30 @@ protected void addInitContainer(PodTemplateSpecBuilder builder, JSONObject initC
         String initContainerAnnotation = builder.buildMetadata().getAnnotations().get(INIT_CONTAINER_ANNOTATION);
         JSONArray initContainers = Strings.isNullOrBlank(initContainerAnnotation) ? new JSONArray()
                 : new JSONArray(initContainerAnnotation);
-        initContainers.put(initContainer);
 
+        // lets avoid duplicates being added
+        boolean duplicate = false;
+        int length = initContainers.length();
+        for (int i = 0; i < length; i++) {
+            JSONObject obj = initContainers.getJSONObject(i);
+            if (duplicateInitcontainer(obj, initContainer)) {
+                duplicate = true;
+            }
+
+        }
+        if (!duplicate) {
+            initContainers.put(initContainer);
+        }
         builder.editMetadata().addToAnnotations(INIT_CONTAINER_ANNOTATION, initContainers.toString()).endMetadata();
     }
+
+    private boolean duplicateInitcontainer(JSONObject a, JSONObject b) {
+        Object name1 = a.get("name");
+        Object name2 = b.get("name");
+        if (Objects.equal(name1, name2)) {
+            log.warn("Found duplicate init containers with names " + name1 + " so ignoring " + b);
```

Comment:
so gogs doesn't include > 1 init container. The issue happens when composing packages together (e.g. cd-pipeline in fabric8-platform). It seems the volume enricher fires again on already enriched resources.

Though any time an enricher goes bad, adding multiple init containers with the same name totally breaks the Deployment. So I'd much rather have f-m-p not totally break the fabric8-platform every time some refactoring happens and an error creeps in. So I'd rather a warning over generating something silently that has no purpose.

I'm more than happy for you to fix the root cause though & feel free to polish the code ;)

BTW I explicitly log the init container JSON so its easier to grok which enricher caused the duplicate; I can see things like this cropping up over time as different enrichers generate containers whose names clash